airflow/cli/cli_parser.py

```diff
@@ -55,6 +56,28 @@ def command(*args, **kwargs):
     return command
 
 
+class DefaultHelpParser(argparse.ArgumentParser):
+    """CustomParser to display help message"""
+    custom_msg = {'celery': 'celery command cab be used in CeleryExcutor'}
+
+    def _check_value(self, action, value):
+        """Override _check_value and check conditionally added command"""
+        if action.choices is not None and value not in action.choices:
+            args = {'value': value,
```

Comment:
```suggestion
    if value in self.custom_msg:
        msg = self.custom_msg[value]
    else:
        args = {
            'value': value,
            'choices': ', '.join(map(repr, action.choices))
        }
        msg = gettext('invalid choice: %(value)r (choose from %(choices)s)') % args
```
It is not necessary to generate a generic message in all cases.  If we have a special message, we display it. When we don't have it, we create it.

Reply:
I agree